js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,72 @@
+describe("enhancedDate Test", function () {
+
+    describe("Testing setDate method", function () {
+        it("should be an instance of a Date object", function () {
+            expect(enhancedDate.setDate()).is.an.instanceof(Date);
+        });
+        it("should equal to now", function () {
+            expect(enhancedDate.setDate().getTime()).to.deep.equal(Date.now());
```

Comment:
This test will frequently fail. When you call setDate().getTim(), it's creating a new Date using "now" as the basis of that Date, then returning the millisecond value of that date. Because this code executes before the code is executed to generate the value it is compared against, Date.now(), there will be a small amount of time (could be only a millesecond, but it's still a different value) difference. If you want to guarantee you're testing the same value, give it an explicit date for each value that gets compared.